pkg/logql/syntax/ast.go

```diff
@@ -53,11 +53,15 @@ type implicit struct{}
 func (implicit) logQLExpr() {}
 
 // LogSelectorExpr is a LogQL expression filtering and returning logs.
+//
+//sumtype:decl
 type LogSelectorExpr interface {
 	Matchers() []*labels.Matcher
-	LogPipelineExpr
+	Pipeline() (Pipeline, error)
 	HasFilter() bool
 	Expr
+
+	isLogSelectorExpr()
```

Comment:
Is this still required? I don't see it in use anywhere

Reply:
This seals the interface and makes it easier to determine which expression implements it. The Protobuf generation is doing something similar for `oneof` and the expressions themselves in our syntax packages does so as well.

Reply:
can we include a comment here that explains this? or is there something I'm missing somewhere else that makes it more obvious